### PR TITLE
phanyzewski/poisson distribution

### PR DESCRIPTION
**Pull request by phanyzewski**
- fix: do not override top level seed
- feat: add gonum module
- fix: update to latest slog api
- fix: add test for validating speed test intervals
- feat: use poisson instead of exponential distribution to prevent too frequent speed testing
- fix: update name to be more reflective of function usage
- chore: update comment
- address security dependency updates
- fix: use default source, create distribution based on minutes not seconds to create more variability in test frequency
- feat: add test covering the minimum number of tests, change frequency to every 4 hours

### CODE REVIEW

File: helpers.go [hunk 1]
```diff
@@ -1,24 +1,28 @@
 package main
 
 import (
-	"math/rand"
 	"time"
 
+	// xrand "golang.org/x/exp/rand"
```
Comment (stevenaldinger) [inline helpers.go:6]:
```suggestion
```
wasn't sure if this was one of those odd things like cgo that was helping with something or just from tinkering

Comment (phanyzewski) [reply helpers.go]: Good catch, I was reading up on some of the exp/rand changes to seed sourcing and tested it out, I did not any significant difference so figured it'd be better to stick with the standard lib here.
